spec/controllers/api/predictions_controller_spec.rb

```diff
@@ -1,113 +1,174 @@
-          response.response_code.should == 422
-        end
-
-        it 'should respond with error messages' do
-          post :create, prediction: @prediction, api_token: @user.api_token
-          response.body.should include('a probability is between 0 and 100%')
-        end
+        specify { expect(response).to_not be_success }
+        specify { expect(response.body).to include('a probability is between') }
       end
     end
 
     context 'with invalid API token' do
-      before(:each) do
-        @prediction = {
-          description: 'The world will end tomorrow!',
-          deadline: 1.day.ago,
-          initial_confidence: '100'
-        }
+      before do
+        post :create, api_token: 'fake-token', prediction: prediction_params
+      end
+
+      specify do
+        expect(response.body).to_not include(prediction_params[:description])
+      end
+      specify { expect(response).to_not be_success }
+    end
+  end
+
+  describe 'PUT /predictions/:id' do
+    let(:new_prediction_params) do
+      { description: 'The world definitely will not end tomorrow!' }
+    end
+
+    before do
+      @user = valid_user(api_token: token)
+      @user.save
+      @prediction = valid_prediction(creator: @user)
+      @prediction.save
+    end
+
+    context 'with valid API token' do
+      context 'authorized user' do
+        before do
+          put :update,
+              api_token: @user.api_token,
+              id: @prediction.id,
+              prediction: new_prediction_params
+        end
+
+        specify { expect(response).to be_success }
+        specify { expect(response.content_type).to eq(Mime::JSON) }
+
+        it 'updates the existing prediction' do
+          description = new_prediction_params[:description]
+          @prediction.reload
+
+          expect(@prediction.description).to eq(description)
+        end
       end
 
-      it 'should not create a new prediction' do
-        post :create, api_token: 'fake-token', prediction: @prediction
-        response.body.should_not include(@prediction[:description])
+      context 'unauthorized user' do
+        before do
+          put :update,
+              api_token: 'fake-token',
+              id: @prediction.id,
+              prediction: new_prediction_params
+        end
+
+        specify { expect(response).to_not be_success }
+        specify { expect(response.content_type).to eq(Mime::JSON) }
       end
+    end
 
-      it 'should respond with HTTP failure' do
-        post :create, api_token: 'fake-token', prediction: @prediction
-        response.response_code.should == 401
+    context 'with invalid API token' do
+      before do
+        put :update,
+            api_token: 'fake-token',
+            id: @prediction.id,
+            prediction: new_prediction_params
       end
+
+      specify { expect(response).to_not be_success }
+      specify { expect(response.content_type).to eq(Mime::JSON) }
     end
   end
 end
```

Comment:
A couple of things with the tests.

In the top half you use let(:var) and in the bottom half you use a mix of ivars and lets. Is the a particular reason for this?
Secondly, the show spec is very stub heavy. Was this intentional or were you keeping them consistent with the other GET spec?